pkg/kubelet/cm/qos_container_manager_linux.go

```diff
@@ -0,0 +1,266 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cm
+
+import (
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/kubelet/qos"
+)
+
+type QOSContainerManager interface {
+	Start(*v1.Node, func() []*v1.Pod) error
+	GetQOSContainersInfo() QOSContainersInfo
+	UpdateReserve() error
```

Comment:
there may be a better name for this, maybe update this to reflect its enforcing the allocatable value at the qos tier.

how about `UpdateQOSAllocatable()`  since we will plug cpu shares in here always so burstable tiers get proportionate guarantees w/ requests.